### DISCUSSION TITLE
How to use Skeleton UI in Svelte (without Sveltekit)

### DISCUSSION THREAD

**Discussion by josiah-redjade** [skeletonlabs/skeleton#1221] [Q&A]
In the documentation, I can only find examples on how to use Skeleton UI with SvelteKit. However, the project I am working on uses vanilla Svelte. Is it still possible to use, or is it only possible to use Skeleton UI with SvelteKit?

Many thanks,
Josiah

**Comment by Sarenor** [comment]
It should work, but we don't officially support it. So if you run into a problem that we can't replicate with SvelteKit, it'll be a low priority issue or you might be on your own.

**Comment by endigo9740** [comment]
Support for Astro and Vite+Svelte (Svelte without SvelteKit) ended in December:
- https://github.com/skeletonlabs/skeleton/discussions/721

While we don't support this, it doesn't mean it's not possible. Everything (to our knowledge) still works with this configuration - but you'll be on your own to implement and test things.

**Comment by cope** [comment]
OK, but... why?

A UI toolkit should not really require anything other than the frontend side of things?

**Reply by endigo9740** [reply]
We explained our reasoning for this in the post linked above:
- https://github.com/skeletonlabs/skeleton/discussions/721

It's a matter of logistics and team size primarily, not a technical limitation. If you're interested in seeing support for this you could consider contributing to the project rather than downvoting comments indiscriminately. This is how open source works :)

Also technically Skeleton still works when using Vite/Svelte or Astro, we just don't officially provide support for it.

**Reply by ArmantG** [reply]
I would like to revisit this now that the documentation site is being moved to Astro? Any plans to have a integration for skeleton ui and astro then? I mean, if the doc site is an astro site but components are svelte, you should be thinking of some kind of setup and documentation for how to use Astro, Svelte and Skeleton?

**Reply by endigo9740** [reply]
@ArmantG I'd recommend you read our full announcement and proposal on the scale and scope of changes coming in Skeleton v3:

- https://github.com/skeletonlabs/skeleton/discussions/2375

We aren't simply moving the doc site to Astro, but fully enabling Skeleton to be framework agnostic. As well as provide components for Svelte 5 and React at launch, eventually expanding to other UI frameworks.

We do plan to have official documentation for SvelteKit and Next.js at launch, but will plan to expand that to others over time. Astro probably won't see a dedicated component package, but given it's support for so many others, it really doesn't need to. That said, there will likely be a few useful components we build for Astro (such as a Table of Contents) and those will be open source as part of the docs.

**Reply by ArmantG** [reply]
Sounds really good to me! Cant wait. Been my favourite Component UI library so looking forward to all the great new things!
